### ISSUE TITLE
useEntitesInfo triggers a useSelect memoization warning

### ISSUE THREAD

**Issue by jsnajdr** [WordPress/gutenberg#56497]
When navigating around the Site Editor (`/wp-admin/site-editor.php?path=/wp_template`) I see console warnings like this:
```
The 'useSelect' hook returns different values when called with the same state and parameters. This can lead to unnecessary rerenders. 
    at NewTemplate (http://localhost:8888/wp-content/plugins/gutenberg/build/edit-site/index.min.js?ver=824ea2c2eee98e4bba76:3592:3)
    at AddNewTemplate (http://localhost:8888/wp-content/plugins/gutenberg/build/edit-site/index.min.js?ver=824ea2c2eee98e4bba76:3425:3)
```

This happens in the `useEntitiesInfo` hook which has a `useSelect` hook that returns an object in this shape:
```js
{
  page: {
    existingEntitiesIds: [],
    hasEntities: false
  }
}
```
Here the inner object (value of the `page` field) is constructed on each call, it's not memoized. Also the `existingEntitiesIds` fallback value (`[]`) should be extracted into a constant, but doing only this doesn't fix the problem.

Fixing this is a rather complex memoization puzzle, so for now I'm just reporting the issue. Maybe someone like @Mamaduka will be interested in working on it and will get to it earlier than me 🙂

**Comment by Mamaduka** [comment]
Thanks for creating an issue, @jsnajdr!

This was only a to-do list somewhere. I'll try to work on it when I've some free time.

**Comment by Mamaduka** [comment]
I believe this was fixed by #60586.

**Comment by Mamaduka** [comment]
Closing. #60586 was merged, and I no longer see the warnings when navigating the site editor.
